source/includes/connection-snippets/x509-snippets/go-connection.go

```diff
@@ -0,0 +1,33 @@
+// begin x509 connection
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Trainer struct {
+    Name string
+    Age  int
+    City string
+}
+
+func main() {
+	ctx := context.TODO()
+	caFilePath := "/etc/certs/mongodb/ca.pem"
+	certificateKeyFilePath := "/etc/certs/mongodb/client.pem"
+
+	uri := "mongodb+srv://<your-x509-client-subject>@<cluster-url>/?tlsCAFile=%s&tlsCertificateKeyFile=%s&authMechanism=MONGODB-X509"
+	uri = fmt.Sprintf(uri, caFilePath, certificateKeyFilePath)
+	clientOpts := options.Client().ApplyURI(uri).SetServerSelectionTimeout(2 * time.Second)
```

Comment:
Why is `SetServerSelectionTimeout` called here? I don't see it being set in the Java and Node examples.

Reply:
Sorry about that, you're right `SetServerSelectionTimeout` doesn't need to be in here. I've removed it now.